File: cpp/examples/move.cpp
/* A simple example showing how to move a cube and print it in H48 format. */

#include "../nissy.h"
#include <iostream>

int main() {
	nissy::cube c;
	if (!c.move("R' U' F").ok()) {
		std::cout << "Error moving the cube!" << std::endl;
		return 1;
	}

	auto str = c.to_string();
	std::cout << "Cube after R' U' F:" << std::endl << str << std::endl;

	return 0;
}

File: python/examples/move.py
# Small example of nissy Python module usage
# See the solve.py example for more details on how this works

import os, sys

# Assume we are eithe rin the top-level directory of the nissy-core repo,
# or in the python subdirectory
sys.path.append(os.getcwd())
sys.path.append(os.getcwd() + os.path.sep + "python")
import nissy

cube = nissy.applymoves(nissy.solved_cube, "R' U' F");
print("Cube after the moves R' U' F:")
print(cube)
